spectree/plugins/falcon_plugin.py

```diff
@@ -167,27 +170,59 @@ def parse_path(self, route, path_parameter_descriptions):
 
         return f'/{"/".join(subs)}', parameters
 
-    def request_validation(self, req, query, json, headers, cookies):
+    def parse_field(self, field):
+        if isinstance(field, list):
+            return [self.parse_field(subfield) for subfield in field]
+
+        # When file name isn't ascii FieldStorage will not consider it.
+        encoded = field.disposition_options.get("filename*")
+        if encoded:
+            encoding, filename = encoded.split("''")
+            field.filename = filename
+            field.file = BytesIO(field.file.read().encode(encoding))
+        if getattr(field, "filename", False):
+            return field
+
+        # This is not a file, thus get flat value (not FieldStorage instance).
+        return field.value
+
+    def request_validation(self, req, query, json, form, headers, cookies):
         if query:
             req.context.query = query.parse_obj(req.params)
         if headers:
             req.context.headers = headers.parse_obj(req.headers)
         if cookies:
             req.context.cookies = cookies.parse_obj(req.cookies)
-        try:
-            media = req.media
-        except self.FALCON_HTTP_ERROR as err:
-            if err.status not in self.FALCON_MEDIA_ERROR_CODE:
-                raise
-            media = None
         if json:
+            try:
+                media = req.media
+            except self.FALCON_HTTP_ERROR as err:
+                if err.status not in self.FALCON_MEDIA_ERROR_CODE:
+                    raise
+                media = None
             req.context.json = json.parse_obj(media)
+        elif form:
+            # This must be done to avoid a bug in cgi.FieldStorage.
+            # https://gist.github.com/kgriffs/fe371bfbb2d63211889b
+            req.env.setdefault("QUERY_STRING", "")
+
+            try:
+                _form = cgi.FieldStorage(fp=req.stream, environ=req.env)  # TODO: rename
+            except ValueError:  # Invalid boundary?
```

Comment:
Will [`falcon.Request.bounded_stream`](https://falcon.readthedocs.io/en/stable/api/request_and_response_wsgi.html?highlight=file#falcon.Request.bounded_stream) be better to avoid this exception?